functions/v2/typed/greeting/main_test.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import main
+
+
+def test_greeting():
+    expected = main.GreetingResponse("Hello Jane Doe!")
+
+    request = {
+        "first_name": "Jane",
+        "last_name": "Doe",
+    }
+
+    req = main.GreetingRequest.from_dict(request)
```

Comment:
These tests do not use the Typed payload parsing logic from FF, and I think they should. Doing so would help us ensure that these samples continue to work correctly for our users.

Based on the [ff tests](https://github.com/GoogleCloudPlatform/functions-framework-python/blob/main/tests/test_typed_event_functions.py#L46-L48), it looks like we just need to do something like this:

```
from functions_framework import create_app
tc = create_app("greet", "./main.py").test_client()
resp = tc.post("/", data=request)
```

Can you try that out for both of these tests?

Reply:
Good point, done. Updated test to exercise the parsing logic in functions framework.